crates/wasmtime/src/runtime/vm/mmap_vec.rs

```diff
@@ -20,13 +22,20 @@ use std::fs::File;
 /// are typically not, then the remaining bytes in the final page for
 /// mmap-backed instances are unused.
 ///
-/// Note that when `signals-based-traps` is disabled then this type is backed
-/// by a normal `Vec<u8>`. In such a scenario this type does not support
-/// read-only or executable bits and the methods are not available.
+/// Note that when `signals-based-traps` is disabled then this type is
+/// backed by the regular memory allocator via `alloc` APIs. In such a
+/// scenario this type does not support read-only or executable bits
+/// and the methods are not available. However, the `CustomCodeMemory`
+/// mechanism may be used by the embedder to set up and tear down
+/// executable permissions on parts of this storage.
 pub enum MmapVec {
     #[doc(hidden)]
     #[cfg(not(feature = "signals-based-traps"))]
-    Vec(Vec<u8>),
+    Alloc {
+        base: base_ptr::BasePtr,
+        len: usize,
```

Comment:
I think `len` can be dropped here in favor of `layout.size()`?

Reply:
Done!